packages/image_picker/image_picker/example/ios/RunnerUITests/ImagePickerFromGalleryUITests.m

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <XCTest/XCTest.h>
+#import "RunnerUITestUtils.h"
+
+@interface ImagePickerFromGalleryUITests : XCTestCase
+
+@property(nonatomic, strong) XCUIApplication* app;
+
+@end
+
+@implementation ImagePickerFromGalleryUITests
+
+- (void)setUp {
+  // Delete the app if already exists, to test permission popups
+
+  self.continueAfterFailure = NO;
+  self.app = [[XCUIApplication alloc] init];
+  [self.app launch];
+  [self addUIInterruptionMonitorWithDescription:@"Permission popups"
+                                        handler:^BOOL(XCUIElement* _Nonnull interruptingElement) {
+                                          XCUIElement* ok = interruptingElement.buttons[@"OK"];
+                                          if (ok.exists) {
+                                            [ok tap];
+                                          }
+                                          // iOS 14.
+                                          XCUIElement* allPhotoPermission =
+                                              interruptingElement
+                                                  .buttons[@"Allow Access to All Photos"];
+                                          if (allPhotoPermission.exists) {
+                                            [allPhotoPermission tap];
+                                          }
+                                          return YES;
+                                        }];
+}
+
+- (void)testPickingFromGallery {
+  [self launchPickerAndCancel];
+  [self launchPickerAndPick];
+}
+
+- (void)launchPickerAndCancel {
+  // Find and tap on the pick from gallery button.
+  NSPredicate* predicateToFindImageFromGalleryButton =
+      [NSPredicate predicateWithFormat:@"label == %@", @"image_picker_example_from_gallery"];
+
+  XCUIElement* imageFromGalleryButton =
+      [self.app.otherElements elementMatchingPredicate:predicateToFindImageFromGalleryButton];
+  if (![imageFromGalleryButton waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find image from gallery button with %@ seconds", @(30));
+  }
+
+  XCTAssertTrue(imageFromGalleryButton.exists);
+  [imageFromGalleryButton tap];
+
+  // Find and tap on the `pick` button.
+  NSPredicate* predicateToFindPickButton =
+      [NSPredicate predicateWithFormat:@"label == %@", @"PICK"];
+
+  XCUIElement* pickButton = [self.app.buttons elementMatchingPredicate:predicateToFindPickButton];
+  if (![pickButton waitForExistenceWithTimeout:30]) {
+    NSLog(@"%@", self.app.debugDescription);
+    XCTFail(@"Failed due to not able to find pick button with %@ seconds", @(30));
```

Comment:
If you're going to hard code it anyway, can do @"Failed due to not able to find pick button with 30 seconds".  Or format as an int.  Don't need the `NSNumber` here but if you do use it, number literals don't need the parentheses, `@30`.

Reply:
done